src/MeshManager.cpp

```diff
@@ -227,11 +227,27 @@ namespace blitzdg {
             }
         }
 
-
         buildBCTable(lines);
         buildConnectivity();
     }
 
+    void MeshManager::buildBCTable(index_type tagNumber) {
+        index_vector_type& E2E = *EToE;
+        index_vector_type& BCTable = *BCType;
+
+        blitz::firstIndex ii;
+        BCTable = 0*ii;
+        for (index_type k=0; k < NumElements; ++k) {
+            for (index_type f=0; f < 3; ++f) {
+                index_type k2 = E2E(k*NumFaces + f);
+
+                // self-referencing elements are boundary elements.
+                if (k2 == k)
+                    BCTable(3*k + f) = tagNumber;
+            }
```

Comment:
Is the magic number 3 equal to `NumFaces` here? If so then you can replace the nested loops with a single loop which would be more efficient. For the test `k2 == k` you could use `k2 == k / NumFaces`, where `k` now ranges over `0,...,NumFaces*NumElements-1`. Even better would be to replace the entire thing with std::copy_if.